src/TeamSpeak3QueryApi/Query/QueryClient.cs

```diff
@@ -39,22 +55,26 @@ public class QueryClient : IDisposable
         private readonly Queue<QueryCommand> _queue = new Queue<QueryCommand>();
         private readonly ConcurrentDictionary<string, List<Action<NotificationData>>> _subscriptions = new ConcurrentDictionary<string, List<Action<NotificationData>>>();
 
+        SshClient _sshClient;
```

Comment:
I think a better abstraction would be creating derived classses of a common base class.

This would also prevent having the wrong default port for different protocols.

Reply:
Confirmed. But we should do this as separate pr?